file_download.php

```diff
@@ -198,7 +198,12 @@
 	'application/pdf' );
 $t_mime_force_attachment = array( 'application/x-shockwave-flash', 'text/html' );
 
-$t_mime_type = substr( $t_content_type, 0, strpos( $t_content_type, ';' ) );
+# extract mime type from content type
+if( strpos( $t_content_type, ';' ) === false ) {
+        $t_mime_type = $t_content_type;
+} else {
+	$t_mime_type = substr( $t_content_type, 0, strpos( $t_content_type, ';' ) );
+}
```

Comment:
I think it would be more readable to get the strpos return value in a $t_semicolon_position variable and re-use it when truncating if `;` was found.